src/backend/joanie/core/admin.py

```diff
@@ -631,6 +658,50 @@ def get_actions(self, request):
         """Remove the dropdown bar menu of action in list view"""
         return {}
 
+    def get_change_actions(self, request, object_id, form_url):
+        """
+        We only need the custom actions to be present in the change form. Each action is
+        present depending on the state of the batch order. Only the action cancel is enabled
+        at any state.
+        """
+        actions = super().get_change_actions(request, object_id, form_url)
+        actions = list(actions)
+
+        if not self.model.objects.filter(
```

Comment:
Here it seems you are querying the database multiple times to retrieve the same batch order.
Wouldn't it be more efficient to fetch the batch order once using its object_id, and then evaluate its state against the enums?